daml/Tests/Distribution/Syndication/Issuance.daml

```diff
@@ -92,28 +94,42 @@ issuance Parties{..} Origination{..} = do
     dateOfTrade = dateOfSettlement
 
   -- Step 1
+  now <- getTime
+  let
+    secret = "mysecret"
+    hashlock = sha256 secret
+    expiry = addRelTime now $ hours 1
   dvpCid11 <- submit operator do exerciseCmd tranche1Cid Structuring.InstructIssuerSettlement with price; dateOfSettlement; dateOfTrade
   dvpCid12 <- submit operator do exerciseCmd tranche2Cid Structuring.InstructIssuerSettlement with price; dateOfSettlement; dateOfTrade
-  dvpCid13 <- submit operator do exerciseCmd tranche3Cid Structuring.InstructIssuerSettlement with price; dateOfSettlement; dateOfTrade
+  dvpCid13 <- submit operator do exerciseCmd tranche3Cid Structuring.InstructIssuerSettlementHtlc with price; dateOfSettlement; dateOfTrade; hashlock; expiry
   allocateInstructions [ issuer, custodian2, settlementBank ]
+  allocateInstructionsHtlc [ issuer, custodian2 ]
   signInstructions [ issuer, custodian2, settlementBank ]
-  settleTrades bondRegistrar [ dvpCid11, dvpCid12, dvpCid13 ]
+  signInstructionsHtlc [ custodian2, settlementBank ]
+  settleTrades bondRegistrar [ dvpCid11, dvpCid12 ]
+  settleTradeHtlc settlementBank (secret, dvpCid13)
 
   -- Step 2
   dvpCid21 <- submit operator do exerciseCmd tranche1Cid Structuring.InstructBndSettlement with price; dateOfSettlement; dateOfTrade
   dvpCid22 <- submit operator do exerciseCmd tranche2Cid Structuring.InstructBndSettlement with price; dateOfSettlement; dateOfTrade
-  dvpCid23 <- submit operator do exerciseCmd tranche3Cid Structuring.InstructBndSettlement with price; dateOfSettlement; dateOfTrade
+  dvpCid23 <- submit operator do exerciseCmd tranche3Cid Structuring.InstructBndSettlementHtlc with price; dateOfSettlement; dateOfTrade; hashlock; expiry
   allocateInstructions [ settlementBank, custodian2, custodian3, bndBank ]
+  allocateInstructionsHtlc [ settlementBank, custodian2, custodian3 ]
   signInstructions [ settlementBank, custodian2, custodian3, bndBank ]
-  settleTrades bondRegistrar [ dvpCid21, dvpCid22, dvpCid23 ]
+  signInstructionsHtlc [ custodian2, custodian3, bndBank ]
+  settleTrades bondRegistrar [ dvpCid21, dvpCid22 ]
+  settleTradeHtlc bndBank (secret, dvpCid23)
 
-  -- -- Step 3
+  -- -- -- Step 3
   dvpCids31 <- submit operator do exerciseCmd tranche1Cid Structuring.InstructInvestorSettlement with confirmationCids = confirmationCids1; price; dateOfSettlement; dateOfTrade
   dvpCids32 <- submit operator do exerciseCmd tranche2Cid Structuring.InstructInvestorSettlement with confirmationCids = confirmationCids2; price; dateOfSettlement; dateOfTrade
-  dvpCids33 <- submit operator do exerciseCmd tranche3Cid Structuring.InstructInvestorSettlement with confirmationCids = confirmationCids3; price; dateOfSettlement; dateOfTrade
+  dvpCids33 <- submit operator do exerciseCmd tranche3Cid Structuring.InstructInvestorSettlementHtlc with confirmationCids = confirmationCids3; price; dateOfSettlement; dateOfTrade; hashlock; expiry
```

Comment:
For investor settlement we need to think who will provide the hashlock. it'll be different for each investor, so maybe it's the investor providing it on the confirmation

Reply:
ah right, each investor should all provide a different hashlock and hold their own secrets.
I will add hashlock into confirmation as commented.
Thanks for the catch.

Reply:
I put an Optional hashlock into Confirmation, probably not the best way.
I was originally thinking of making SettlementMode something like
`SettlementMode = Fop | Dvp | Htlc hashlock: Text` but it doesn't fit well in the current Tranche template and made it harder to work with.